include/boost/geometry/strategy/cartesian/precise_area.hpp

```diff
@@ -0,0 +1,117 @@
+// Boost.Geometry
+
+// Copyright (c) 2020, Oracle and/or its affiliates.
+
+// Contributed and/or modified by Vissarion Fysikopoulos, on behalf of Oracle
+
+// Licensed under the Boost Software License version 1.0.
+// http://www.boost.org/users/license.html
+
+#ifndef BOOST_GEOMETRY_STRATEGY_CARTESIAN_AREA_ACCURATE_HPP
+#define BOOST_GEOMETRY_STRATEGY_CARTESIAN_AREA_ACCURATE_HPP
+
+#include <boost/mpl/if.hpp>
+
+//#include <boost/geometry/arithmetic/determinant.hpp>
+#include <boost/geometry/core/access.hpp>
+#include <boost/geometry/core/coordinate_type.hpp>
+#include <boost/geometry/core/coordinate_dimension.hpp>
+#include <boost/geometry/strategy/area.hpp>
+#include <boost/geometry/util/select_most_precise.hpp>
+#include <boost/geometry/util/precise_math.hpp>
+
+namespace boost { namespace geometry
+{
+
+namespace strategy { namespace area
+{
+
+/*!
+\brief Cartesian area calculation
+\ingroup strategies
+\details Calculates cartesian area using the trapezoidal rule and precise summation
```

Comment:
A minor suggestion: The "precise summation" here kind of implies it already because summation precision is no issue with integral types, but maybe it would be helpful for some users who are less familiar with this to explicitly state that this is only intended to be used for floating-point CalculationType. Because, if I understand area result_type correctly, it will only promote to double if the user did not specify something.

Or maybe there could be a static_assert of std::is_floating_point<return_type> in the strategy (it could also be argued to have it in the precise_math header, but I think from the users perspective, getting that error for the strategy might be more helpful).

Reply:
Yes, if the user passes CalculationType we believe in his intentions and use this type, no matter what it is. So I have no opinion regarding this suggestion.

Have in mind that a user can pass his own, precise numeric type. If this method is good for such types then the test should rather be `(! std::is_integral<return_type>::value)`.

Reply:
I clarify it more in the `\details` for the documentation. Since this is supposed to be for advanced users I do not think we need an assert or return an error. Are you OK with this @tinko92 ?

Reply:
Yes, I think this is OK.